libsignal-service/src/websocket.rs

```diff
@@ -377,19 +377,61 @@ impl SignalWebSocket {
     {
         let response = self.request(r).await?;
         if response.status() != 200 {
-            log::error!(
+            log::debug!(
                 "request_json with non-200 status code. message: {}",
                 response.message()
             );
-            return Err(ServiceError::UnhandledResponseCode {
+        }
+
+        fn json<U>(body: &[u8]) -> Result<U, ServiceError>
+        where
+            for<'de> U: serde::Deserialize<'de>,
+        {
+            serde_json::from_slice(body).map_err(|e| {
+                ServiceError::JsonDecodeError {
+                    reason: e.to_string(),
+                }
+            })
+        }
+
+        match response.status() {
+            200 => json(response.body()),
```

Comment:
nit: `200 | 204`

Reply:
I didn't dare, because I feared the `200 | 204 == 204` because of bitwise or. Do you know which of both options it takes?

Reply:
The left part of each arm of a match expression is a pattern not an expression. That means you can only do pattern matching there. You can't use operators, accessors and function calls.

Will change.